File: drift/src/main/java/drift/com/drift/wrappers/ConversationListWrapper.java
package drift.com.drift.wrappers;


import java.util.ArrayList;


import drift.com.drift.api.APIManager;
import drift.com.drift.helpers.LoggerHelper;

import drift.com.drift.model.ConversationExtra;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * Created by eoin on 04/08/2017.
 */

public class ConversationListWrapper {

    private static String TAG = ConversationListWrapper.class.getSimpleName();

    public static void getConversationsForEndUser(int endUserId, final APICallbackWrapper<ArrayList<ConversationExtra>> callback){

        APIManager.getConversationClient().getConversationsForEndUser (endUserId).enqueue(new Callback<ArrayList<ConversationExtra>>() {
            @Override
            public void onResponse(Call<ArrayList<ConversationExtra>> call, Response<ArrayList<ConversationExtra>> response) {
                if((response.code() != 200 && response.code() != 201) || response.body() == null) {
                    callback.onResponse(null);
                }else{
                    callback.onResponse(response.body());
                }
            }

            @Override
            public void onFailure(Call<ArrayList<ConversationExtra>> call, Throwable t) {
                LoggerHelper.logMessage(TAG, t.getLocalizedMessage());
                callback.onResponse(null);
            }
        });
    }

}

File: drift/src/main/java/drift/com/drift/wrappers/ConversationListWrapper.kt
package drift.com.drift.wrappers


import java.util.ArrayList


import drift.com.drift.api.APIManager
import drift.com.drift.helpers.LoggerHelper

import drift.com.drift.model.ConversationExtra

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

/**
 * Created by eoin on 04/08/2017.
 */

internal object ConversationListWrapper {

    private val TAG = ConversationListWrapper::class.java.simpleName

    fun getConversationsForEndUser(endUserId: Long?, callback: (response: ArrayList<ConversationExtra>?) -> Unit) {

        APIManager.conversationClient.getConversationsForEndUser(endUserId).enqueue(object : Callback<ArrayList<ConversationExtra>> {
            override fun onResponse(call: Call<ArrayList<ConversationExtra>>, response: Response<ArrayList<ConversationExtra>>) {
                if (response.code() != 200 && response.code() != 201 || response.body() == null) {
                    callback(null)
                } else {
                    callback(response.body())
                }
            }

            override fun onFailure(call: Call<ArrayList<ConversationExtra>>, t: Throwable) {
                LoggerHelper.logMessage(TAG, t.localizedMessage)
                callback(null)
            }
        })
    }

}
